File: backend/src/main/java/com/back/ourlog/global/security/exception/JwtAuthenticationException.java
package com.back.ourlog.global.security.exception;

import com.back.ourlog.global.exception.ErrorCode;
import org.springframework.security.core.AuthenticationException;

public class JwtAuthenticationException extends AuthenticationException {

    private final ErrorCode errorCode;

    public JwtAuthenticationException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }

    public JwtAuthenticationException(ErrorCode errorCode, Throwable cause) {
        super(errorCode.getMessage(), cause);
        this.errorCode = errorCode;
    }

    public ErrorCode getErrorCode() {
        return errorCode;
    }
}

File: backend/src/main/java/com/back/ourlog/global/security/exception/JwtAuthenticationException.kt
package com.back.ourlog.global.security.exception

import com.back.ourlog.global.exception.ErrorCode
import org.springframework.security.core.AuthenticationException

class JwtAuthenticationException : AuthenticationException {
    val errorCode: ErrorCode

    constructor(errorCode: ErrorCode) : super(errorCode.message) {
        this.errorCode = errorCode
    }

    constructor(errorCode: ErrorCode, cause: Throwable?) : super(errorCode.message, cause) {
        this.errorCode = errorCode
    }
}
